target/linux/ath79/dts/qca9558_librerouter-v1.dts

```diff
@@ -0,0 +1,213 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "qca9557.dtsi"
+
+/ {
+	compatible = "librerouter,librerouter-v1", "qca,qca9557";
+	model = "LibreRouter v1";
+
+	chosen {
+		bootargs = "console=ttyS0,115200n8";
+	};
+
+	aliases {
+		led-boot = &system;
+		led-failsafe = &system;
+		led-running = &system;
+		led-upgrade = &system;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		system: system {
+			label = "librerouter:green:system";
+			gpios = <&gpio 23 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "heartbeat";
+		};
+
+		wifi_green {
+			label = "librerouter:green:wlan2g";
+			gpios = <&gpio 22 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";
+		};
+
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "Reset button";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+
+	};
+
+	gpio-export {
+		compatible = "gpio-export";
+		#size-cells = <0>;
+
+		gpio_poe_passthrough {
+			gpio-export,name = "librerouter:power:poe_pass";
+			gpio-export,output = <0>;
+			gpios = <&gpio 1 GPIO_ACTIVE_HIGH>;
+		};
+	};
+
+	watchdog {
+		compatible = "linux,wdt-gpio";
+		gpios = <&gpio 2 GPIO_ACTIVE_LOW>;
+		hw_algo = "toggle";
+		hw_margin_ms = <1000>;
+		always-running;
+	};
+
+};
+
+&pcie0 {
+	status = "okay";
+};
+
+&pcie1 {
+	status = "okay";
+};
+
+&uart {
+	status = "okay";
+};
+
+&gpio {
+	status = "okay";
+};
+
+&usb_phy0 {
+	status = "okay";
+};
+
+&usb0 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+	status = "okay";
+
+	hub_port0: port@1 {
+		reg = <1>;
+		#trigger-source-cells = <0>;
+	};
+};
+
+&usb_phy1 {
+	status = "okay";
+};
+
+&usb1 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+	status = "okay";
+
+	hub_port1: port@1 {
+		reg = <1>;
+		#trigger-source-cells = <0>;
+	};
+};
+
+&spi {
+	status = "okay";
+	num-cs = <1>;
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <25000000>;
+
```

Comment:
If this is a SPI-NOR flash (the binding says so), you should definitely try setting the FAST READ property

"m25p,fast-read;"

for a cheap boost. (Please also run some tests).

Reply:
Great! didn't know about fast-read, will try and come back with results.

Reply:
With 25MHz SPI clock m25p,fast-read only slow-down the read operation. Fast read instructions is used for SPI clock higher than 50MHz and the extra dummy byte required by that instruction is a waste of 8 clocks when normal read instruction works fine.
There won't be any differences in ath79 with 4.14 kernel though, since we have a hacky flash_read function to use the memory-mapped area for reading operation.

Reply:
Thanks both, I will leave it without change for now.

Reply:
Ahh, 

> Great! didn't know about fast-read, will try and come back with results.

no results. I want to know, even if they are worse.

Reply:
Tested with 
```
                compatible = "jedec,spi-nor";
                reg = <0>;
                spi-max-frequency = <25000000>;
+               m25p,fast-read;
 
                partitions {
```
No improvement in boot time nor reading from /dev/mtd:
```
# 250000 MHz no fast-read
[   11.747702] kmodloader: done loading kernel modules from /etc/modules.d/*
root@OpenWrt:/# time cat /dev/mtd2 > /dev/null 
real    0m 5.45s
user    0m 0.00s
sys     0m 5.32s
root@OpenWrt:/# time cat /dev/mtd2 > /dev/null 
real    0m 5.45s
user    0m 0.00s
sys     0m 5.30s

# 25000 MHz fast-read
[   11.735793] kmodloader: done loading kernel modules from /etc/modules.d/*
root@OpenWrt:/# time cat /dev/mtd2 > /dev/null 
real    0m 5.45s
user    0m 0.00s
sys     0m 5.34s
root@OpenWrt:/# time cat /dev/mtd2 > /dev/null 
real    0m 5.44s
user    0m 0.01s
sys     0m 5.32s
```